src/libraries/Common/tests/WasmTestRunner/WasmTestRunner.cs

```diff
@@ -30,41 +34,105 @@ public static async Task<int> Main(string[] args)
             switch (option)
             {
                 case "-notrait":
-                    excludedTraits.Add (args[i + 1]);
+                    excludedTraits.Add(args[i + 1]);
                     i++;
                     break;
                 case "-trait":
-                    includedTraits.Add (args[i + 1]);
+                    includedTraits.Add(args[i + 1]);
                     i++;
                     break;
                 case "-namespace":
-                    includedNamespaces.Add (args[i + 1]);
+                    includedNamespaces.Add(args[i + 1]);
                     i++;
                     break;
                 case "-class":
-                    includedClasses.Add (args[i + 1]);
+                    includedClasses.Add(args[i + 1]);
                     i++;
                     break;
                 case "-method":
-                    includedMethods.Add (args[i + 1]);
+                    includedMethods.Add(args[i + 1]);
                     i++;
                     break;
+                case "-mt":
+                    break;
                 default:
                     throw new ArgumentException($"Invalid argument '{option}'.");
             }
         }
 
-        var runner = new SimpleWasmTestRunner()
+        WasmApplicationEntryPointBase? runner = null;
+        if (args.Contains("-mt"))
+        {
+            Console.WriteLine("Using MultiThreadedTestRunner");
+            runner = new MultiThreadedTestRunner()
+            {
+                TestAssembly = testAssembly,
+                ExcludedTraits = excludedTraits,
+                IncludedTraits = includedTraits,
+                IncludedNamespaces = includedNamespaces,
+                IncludedClasses = includedClasses,
+                IncludedMethods = includedMethods
+            };
+        }
+        else
         {
-            TestAssembly = testAssembly,
-            ExcludedTraits = excludedTraits,
-            IncludedTraits = includedTraits,
-            IncludedNamespaces = includedNamespaces,
-            IncludedClasses = includedClasses,
-            IncludedMethods = includedMethods
-        };
+            Console.WriteLine("Using SimpleWasmTestRunner");
+            runner = new SimpleWasmTestRunner()
+            {
+                TestAssembly = testAssembly,
+                ExcludedTraits = excludedTraits,
+                IncludedTraits = includedTraits,
+                IncludedNamespaces = includedNamespaces,
+                IncludedClasses = includedClasses,
+                IncludedMethods = includedMethods
+            };
+        }
 
         await Task.Yield();
-        return await runner.Run();
+        await runner.RunAsync();
+        return runner.LastRunHadFailedTests ? 1 : 0;
```

Comment:
are we loosing different non `1` exit codes here ?

Reply:
https://github.com/dotnet/xharness/blob/dc88df0d8ae2dfa3381611fd5d843ec3e0063cbf/src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmApplicationEntryPoint.cs#L54